File: apps/calenderapp/frontend/src/types/marketEvent.js
/**
 * @typedef {Object} MarketEventFilters
 * @property {string} date_from
 * @property {string} date_to
 * @property {string} keyword
 * @property {string} industry
 * @property {string} theme
 * @property {string} event_type
 * @property {boolean|null} favorites_only
 * @property {boolean|null} is_cross_stock
 * @property {number|null} min_affected_stock_count
 * @property {boolean|null} is_active  DEPRECATED — see kb_market_event schema. Frozen at false for new rows; UI no longer renders the "发酵中" badge that consumed this field. Reserved for redesign.
 */

/**
 * @typedef {Object} MarketEventStock
 * @property {string} stock_code
 * @property {string} stock_name
 */

/**
 * @typedef {Object} MarketEventSourceReport
 * @property {string} report_id
 * @property {string} report_title
 * @property {string} stock_code
 * @property {string} stock_name
 * @property {string} report_date
 * @property {string} source_name
 * @property {string} source_url
 * @property {string} source_path
 */

/**
 * @typedef {Object} MarketEventListItem
 * @property {string} event_key
 * @property {string} event_name
 * @property {string} event_time_text
 * @property {string} event_content
 * @property {string} event_type
 * @property {string} event_scope
 * @property {string} scope_reason
 * @property {string} primary_industry
 * @property {string} primary_theme
 * @property {string[]} affected_industries
 * @property {string[]} affected_themes
 * @property {number} affected_stock_count
 * @property {MarketEventStock[]} affected_stocks_preview
 * @property {number} source_report_count
 * @property {string} first_seen_date
 * @property {string} latest_active_date
 * @property {string[]} active_dates
 * @property {boolean} is_cross_stock
 * @property {boolean} is_active  DEPRECATED — frozen at false for new events. Historical rows preserved.
 * @property {string} review_status
 * @property {string} review_updated_at
 * @property {string} record_source
 * @property {boolean} is_favorite
 */

/**
 * @typedef {Object} MarketEventDetail
 * @property {string} event_key
 * @property {string} event_name
 * @property {string} event_time_text
 * @property {string} event_content
 * @property {string} event_type
 * @property {string} event_scope
 * @property {string} scope_reason
 * @property {string} primary_industry
 * @property {string} primary_theme
 * @property {string[]} affected_industries
 * @property {string[]} affected_themes
 * @property {string} risk_summary
 * @property {string} first_seen_date
 * @property {string} latest_active_date
 * @property {string[]} active_dates
 * @property {MarketEventStock[]} affected_stocks
 * @property {MarketEventSourceReport[]} source_reports
 * @property {string[]} source_event_ids
 * @property {string} review_status
 * @property {string} review_updated_at
 * @property {string} record_source
 * @property {boolean} is_favorite
 */

/**
 * @typedef {Object} MarketEventTimelinePoint
 * @property {string} date
 * @property {number} affected_stock_count
 * @property {MarketEventStock[]} stocks
 * @property {number} source_report_count
 */

/**
 * @typedef {Object} MarketEventReview
 * @property {string} event_key
 * @property {string} review_status
 * @property {string} review_version
 * @property {string} review_source
 * @property {string} event_truth
 * @property {string} time_truth
 * @property {string} content_truth
 * @property {string} disposition
 * @property {number} confidence
 * @property {string} headline
 * @property {string} summary
 * @property {Object} review_payload
 * @property {Object} source_snapshot
 * @property {string} error_message
 * @property {string} requested_at
 * @property {string} completed_at
 * @property {string} created_at
 * @property {string} updated_at
 */

/**
 * @typedef {Object} MarketEventFilterMeta
 * @property {string[]} industries
 * @property {string[]} themes
 * @property {string} date_min
 * @property {string} date_max
 */

export function createDefaultMarketEventFilters(overrides = {}) {
  return {
    date_from: '',
    date_to: '',
    keyword: '',
    industry: '',
    theme: '',
    event_type: 'all',
    favorites_only: null,
    is_cross_stock: null,
    min_affected_stock_count: null,
    is_active: null,
    ...overrides
  }
}

export function createDefaultMarketEventListItem() {
  return {
    event_key: '',
    event_name: '',
    event_time_text: '',
    event_content: '',
    event_type: '',
    event_scope: '',
    scope_reason: '',
    primary_industry: '',
    primary_theme: '',
    affected_industries: [],
    affected_themes: [],
    affected_stock_count: 0,
    affected_stocks_preview: [],
    source_report_count: 0,
    first_seen_date: '',
    latest_active_date: '',
    active_dates: [],
    is_cross_stock: false,
    is_active: false,
    review_status: '',
    review_updated_at: '',
    record_source: 'market',
    is_favorite: false
  }
}

export function createDefaultMarketEventDetail() {
  return {
    event_key: '',
    event_name: '',
    event_time_text: '',
    event_content: '',
    event_type: '',
    event_scope: '',
    scope_reason: '',
    primary_industry: '',
    primary_theme: '',
    affected_industries: [],
    affected_themes: [],
    risk_summary: '',
    first_seen_date: '',
    latest_active_date: '',
    active_dates: [],
    affected_stocks: [],
    source_reports: [],
    source_event_ids: [],
    review_status: '',
    review_updated_at: '',
    record_source: 'market',
    is_favorite: false
  }
}

export function createDefaultMarketEventTimelinePoint() {
  return {
    date: '',
    affected_stock_count: 0,
    stocks: [],
    source_report_count: 0
  }
}

export function createDefaultMarketEventFilterMeta() {
  return {
    industries: [],
    themes: [],
    date_min: '',
    date_max: ''
  }
}

export function createDefaultMarketEventReview() {
  return {
    event_key: '',
    review_status: '',
    review_version: '',
    review_source: '',
    event_truth: '',
    time_truth: '',
    content_truth: '',
    disposition: '',
    confidence: 0,
    headline: '',
    summary: '',
    review_payload: {},
    source_snapshot: {},
    error_message: '',
    requested_at: '',
    completed_at: '',
    created_at: '',
    updated_at: ''
  }
}

File: apps/stockkb/src/stockrag/event_kb/schemas/market_event.py
from __future__ import annotations

from dataclasses import asdict, dataclass, field
from typing import Any


@dataclass(slots=True)
class MarketEventStock:
    stock_code: str
    stock_name: str = ""

    def to_dict(self) -> dict[str, Any]:
        return asdict(self)


@dataclass(slots=True)
class MarketEventSourceReport:
    report_id: str
    report_title: str
    stock_code: str
    stock_name: str
    report_date: str
    source_name: str = ""
    source_url: str = ""
    source_path: str = ""

    def to_dict(self) -> dict[str, Any]:
        return asdict(self)


@dataclass(slots=True)
class MarketEventListItem:
    event_key: str
    event_name: str
    event_time_text: str
    event_type: str = ""
    event_scope: str = ""
    scope_reason: str = ""
    event_content: str = ""
    primary_industry: str = ""
    primary_theme: str = ""
    affected_industries: list[str] = field(default_factory=list)
    affected_themes: list[str] = field(default_factory=list)
    affected_stock_count: int = 0
    affected_stocks_preview: list[MarketEventStock] = field(default_factory=list)
    source_report_count: int = 0
    first_seen_date: str = ""
    latest_active_date: str = ""
    active_dates: list[str] = field(default_factory=list)
    is_cross_stock: bool = False
    # DEPRECATED — frozen at False for new rows; historical data preserved.
    # Old definition: "latest_active_date within 5 days". UI badge removed.
    # See market_event_builder._is_active_date for the historical heuristic.
    is_active: bool = False
    is_favorite: bool = False

    def to_dict(self) -> dict[str, Any]:
        payload = asdict(self)
        payload["affected_stocks_preview"] = [item.to_dict() for item in self.affected_stocks_preview]
        return payload


@dataclass(slots=True)
class MarketEventDetail:
    event_key: str
    event_name: str
    event_time_text: str
    event_content: str
    event_type: str = ""
    event_scope: str = ""
    scope_reason: str = ""
    primary_industry: str = ""
    primary_theme: str = ""
    affected_industries: list[str] = field(default_factory=list)
    affected_themes: list[str] = field(default_factory=list)
    risk_summary: str = ""
    first_seen_date: str = ""
    latest_active_date: str = ""
    active_dates: list[str] = field(default_factory=list)
    affected_stocks: list[MarketEventStock] = field(default_factory=list)
    source_reports: list[MarketEventSourceReport] = field(default_factory=list)
    source_event_ids: list[str] = field(default_factory=list)
    is_favorite: bool = False

    def to_dict(self) -> dict[str, Any]:
        payload = asdict(self)
        payload["affected_stocks"] = [item.to_dict() for item in self.affected_stocks]
        payload["source_reports"] = [item.to_dict() for item in self.source_reports]
        return payload


@dataclass(slots=True)
class MarketEventTimelinePoint:
    date: str
    affected_stock_count: int = 0
    stocks: list[MarketEventStock] = field(default_factory=list)
    source_report_count: int = 0

    def to_dict(self) -> dict[str, Any]:
        payload = asdict(self)
        payload["stocks"] = [item.to_dict() for item in self.stocks]
        return payload


@dataclass(slots=True)
class MarketEventFilterMeta:
    industries: list[str] = field(default_factory=list)
    themes: list[str] = field(default_factory=list)
    date_min: str = ""
    date_max: str = ""

    def to_dict(self) -> dict[str, Any]:
        return asdict(self)
